### DISCUSSION TITLE
Dịch từ "downstream" và "upstream"

### DISCUSSION THREAD

**Discussion by quangvnai** [d2l-ai/d2l-vi#2589] [Terminology]
* Mình đã thêm từ "**downstream**" vào glossary với cách dịch là **hạ nguồn**. Và một cách rất tự nhiên có từ "**upstream**", có thể sẽ được dịch là **thượng nguồn**.

* Như anh @LongNguyen1984 có gợi ý là dịch **downstream** là **xuôi dòng**, và mình đoán có thể sẽ dịch **upstream** là **ngược dòng**?

Mọi người thấy cách dịch nào phù hợp? @lkhphuc @minhduc0711 @rootonchair @quangnhat185 @cuongvng @duythanhvn

**Comment by thanhcsf** [comment]
Thật là ... thuỷ điện..

**Reply by thanhcsf** [reply]
Mình thấy gợi ý anh Long ổn với mình. Mà từ stream thì mình hay dùng là **luồng**, có nên đổi chút cho nó "khoa học kỹ thuật" thêm tí... hạ thượng xuôi ngược thuỷ điện ghê... 😅

**Reply by quangvnai** [reply]
lần trước còn tra từ điển ra cách dịch là "hạ lưu". Từ này thì max thuỷ lợi :)))

**Comment by rootonchair** [comment]
Theo post này, https://www.reddit.com/r/LanguageTechnology/comments/g1c8yt/what_does_downstream_tasks_mean_in_nlp/

thì đây là tác vụ ở cuối pipeline ML của mình, nhận input là output của một tác vụ khác. Vậy nên từ `dòng` em thấy hợp lý

**Reply by quangvnai** [reply]
Yeah, từ `dòng` ở đây make sense. Đợi thêm 1 2 ý kiến rồi anh sửa luôn hai từ này ;)

**Reply by minhduc0711** [reply]
Nếu upstream/downstream là đầu/cuối pipeline thì `hạ nguồn` hay `hạ lưu` nghe hợp lý hơn chứ nhỉ. `Xuôi/ngược dòng` làm mình nghĩ đến hướng của cái gì đó hơn.

**Reply by cuongvng** [reply]
em thấy ổn với `dòng`

**Comment by quangvnai** [comment]
Hiện tại số đông vote cho **xuôi/ngược dòng**. Em thấy từ **dòng** cũng khá hợp lý. Vậy để em add từ này vào glossary cho mọi ng đỡ đợi ở các PR khác. PR mới tại [đây](https://github.com/aivivn/d2l-vn/pull/2620) nhé mọi người.
Thanks cả nhà.
